src/ConstantProductFactory.sol

```diff
@@ -73,4 +107,16 @@ contract ConstantProductFactory {
         order = amm.getTradeableOrder(tradingParams);
         signature = abi.encode(order, tradingParams);
     }
+
+    /**
+     * @notice Deposit sender's funds into the the AMM contract, assuming that
+     * the sender has approved this contract to spend both tokens.
+     * @param amm the AMM where to send the funds
+     * @param amount0 amount of AMM's token0 to deposit
+     * @param amount1 amount of AMM's token1 to deposit
+     */
+    function deposit(ConstantProduct amm, uint256 amount0, uint256 amount1) public {
+        amm.token0().safeTransferFrom(msg.sender, address(amm), amount0);
+        amm.token1().safeTransferFrom(msg.sender, address(amm), amount1);
+    }
```

Comment:
Makes sense, Do you think it's ok to assume that no user can deploy the same AMM for the same tokens more than once? I can't think of a reason why you would, but I want to make sure I'm not missing something. Otherwise we'd need a "nonce" parameter in the creation function.

In any case, I'd handle this in another PR and merge this without changing the deployment mechanism.

Reply:
Yep, I think it's a reasonable limitation to put on when weighing the risks vs some obscure use case.

Reply:
Deterministic deployments have been implemented [here](https://github.com/cowprotocol/cow-amm/pull/56).